Makefile.uk

```diff
@@ -11,6 +11,8 @@ COMPFLAGS-$(call have_gcc)	+= -fno-tree-sra -fno-split-stack -nostdinc
 ifneq ($(HAVE_STACKPROTECTOR),y)
 COMPFLAGS    += -fno-stack-protector
 endif
+COMPFLAGS    += -fcf-protection=none
```

Comment:
```suggestion
ifneq ($(CONFIG_OPTIMIZE_CFC),y)
COMPFLAGS    += -fcf-protection=none
endif
```

Wouldn't it be better if we added a config option for this? And have the C++ libraries enable that flag automatically? 

Basically, we would have the `-fcf-protection=none` missing for all apps/libs, besides cpp ones, which will disable the optimization.

Reply:
I agree that cf-protection should eventually become a Kconfig build option and have smth like this in the Makefile
```
ifeq ($(CONFIG_OPTIMIZE_CFC),y)
COMPFLAGS += -fcf-protection=full
else
COMPFLAGS += -fcf-protection=none
endif
```
(although I'd rather not use "OPTIMIZE" for build flags that do security hardening => "CONFIG_HARDEN_CFC" or similar)

However:
- `-fcf-protection` sets the `__CET__` macro which any code can check for. This is not limited to C++ unwinders, but arbitrary code that is aware of control flow protections (e.g. off the top of my head: JIT compilers that need to know to emit `endbr64` at the beginning of generated functions).
- if `__CET__` is defined, the header `<cet.h>` is expected to be provided. Unfortunately at this time unikraft cannot provide compiler headers in its include path (and providing them without also pulling in system includes is not exactly trivial) so any CET-aware code will fail to compile.
- even if we somehow provide `<cet.h>`, since CET is a hardware feature (well, two features: SHSTK and IBT), proper use of it requires explicit support code in the "kernel" (i.e., unikraft core) in addition to compiler flags. We do not yet have this code so any incidental use of cf-protection does not actually add to security (and might even break things if e.g., CET-aware code tries to access a non-existent shadow stack).

Due to these issues I consider that force-disabling cf-protection is the right call at this time, with proper implementation in the form of (1) a user-selectable Kconfig option and (2) support code in unikraft core to come in the future.

Reply:
@mariasfiraiala, please add an issue that points out the issue you raised and @andreittr's input for a future solution with the 3 steps above.